src/SprykerEco/Zed/Braintree/Communication/Plugin/Checkout/BraintreeCheckoutDoSaveOrderPlugin.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Braintree\Communication\Plugin\Checkout;
+
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\SaveOrderTransfer;
+use Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutDoSaveOrderInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin as BaseAbstractPlugin;
+
+/**
+ * @method \SprykerEco\Zed\Braintree\Business\BraintreeFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Braintree\Persistence\BraintreeQueryContainerInterface getQueryContainer()
+ * @method \SprykerEco\Zed\Braintree\BraintreeConfig getConfig()
+ * @method \SprykerEco\Zed\Braintree\Communication\BraintreeCommunicationFactory getFactory()
+ */
+class BraintreeCheckoutDoSaveOrderPlugin extends BaseAbstractPlugin implements CheckoutDoSaveOrderInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Saves order payment method data according to quote and checkout response transfer data.
+     * - Used together with {@link \SprykerEco\Zed\Braintree\Communication\Plugin\Checkout\BraintreeCheckoutPostSavePlugin} and {@link \SprykerEco\Zed\Braintree\Communication\Plugin\Checkout\BraintreeCheckoutPreConditionPlugin}
```

Comment:
what is the reason to specify this?

Reply:
As discussed on Slack, in this PR 4 old plugins were deprecated and 3 new ones were introduced. All the 3 new plugins must be used together. **The news plugins can't be mixed with the deprecated ones**. Otherwise, payment just won't work.